src/Storages/MergeTree/MergeTreeMarksLoader.cpp

```diff
@@ -167,6 +167,16 @@ MarkCache::MappedPtr MergeTreeMarksLoader::loadMarksImpl()
         }
     }
 
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
```

Comment:
`if constexpr (std::endian::native == std::endian::big)`

Reply:
I'd prefer this myself, but unfortunately it causes the compiler to generate some warnings that then turn into errors.